apps/designer/app/shared/dom-hooks/use-measure.ts

```diff
@@ -19,11 +20,33 @@ export const useMeasure = <
   const [element, setElement] = useState<MeasuredElement | null>(null);
   const [rect, setRect] = useState<DOMRect>();
 
+  const [isInline, setIsInline] = useState(false);
+
   const handleChange = useCallback(() => {
     if (element === null) return;
-    setRect(element.getBoundingClientRect());
+    setIsInline(window.getComputedStyle(element).display === "inline");
+    const nextRect = element.getBoundingClientRect();
+    setRect((currentRect) => {
+      if (
+        currentRect === undefined ||
+        currentRect.x !== nextRect.x ||
+        currentRect.y !== nextRect.y ||
+        currentRect.width !== nextRect.width ||
+        currentRect.height !== nextRect.height
+      ) {
+        return nextRect;
+      }
+      return currentRect;
+    });
   }, [element]);
 
+  // ResizeObserver does not work for inline elements,
+  // so if the element is inline, we measure it after any change in the instance tree.
+  //
+  // NOTE: We assume useMeasure is used only to measure elements on canvas.
```

Comment:
use-measure is in generic dom-hooks, so no, lets split up the generic part that can keep living here and the canvas specific part that would then move to canvas feature folder